core/views.py

```diff
@@ -130,24 +132,81 @@ def _wrapped(request, *args, **kwargs):
     return _wrapped
 
 
-def index(request):
-    collections = Collection.objects.filter(owner=request.user)
-    all_contents = Content.objects.filter(collection__in=collections)
-    filtered_contents = {collection: [] for collection in collections}
+def prepare_collection_for_display(collection):
+    published_contents = Content.objects.filter(collection=collection).filter(
+        published=True
+    )
+    contents_count = published_contents.count()
+    parsed_collection = {
+        "name": collection.name,
+        "items_display": f"{contents_count} items"
+        if contents_count > 1 or contents_count == 0
+        else f"{contents_count} item",
```

Comment:
Not that it matters that much, but I'm surprised `ruff` didn't want to indent these if/else lines more.